src/bonds.cpp

```diff
@@ -288,6 +288,48 @@ void PeriodicDihedral::setEnergyFunction(const ParticleVector& particles) {
         const auto angle = std::atan2((norm1.cross(norm2)).dot(v2) / v2.norm(), norm1.dot(norm2));
         return force_constant * (1.0 + std::cos(periodicity * angle - dihedral_angle));
     };
+    forceFunc = [&](Geometry::DistanceFunction dist) -> std::vector<IndexAndForce> {
+        // Torsion on the form a(0) - b(1) - c(2) - d(3)
+        // The need to recalclate the dihedral angle, using the energy function implementation.
+        auto v1 = dist(particles[indices[1]].pos, particles[indices[0]].pos); // a->b
+        auto v2 = dist(particles[indices[2]].pos, particles[indices[1]].pos); // b->c
+        auto v3 = dist(particles[indices[3]].pos, particles[indices[2]].pos); // c->d
+        auto norm1 = v1.cross(v2);                                            // ab x bc
```

Comment:
`auto` -> `point` or add `.eval()` after `.cross(v3)`

Reply:
`v1.cross(v2)` is used twice in the code, so better to save it. Also I think the name `normX` should be reserved for the length of a vector. Try to be explicit by e.g. `normal` or simular for cross products.

Reply:
Fixed with 8ff5d87.